### ISSUE TITLE
Angular 13 and MFE breaks with Cypress

### ISSUE THREAD

**Issue by smasala** [nrwl/nx#8081]
## Current Behavior

The `styles` defined in `angular.json` cannot be injected `default value = inject:true`.
Causes the test to fail completely.

## Steps to Reproduce
1. Create a **fresh** nx workspace
2. add an angular app with mfe preconfigured
3. run e2e tests.

The nx custom webpack config breaks the tests. Remove the custom webpack config and the e2e tests work again.

### Failure Logs

```
       "before each" hook for "should display welcome message":
     SyntaxError: The following error originated from your application code, not from Cypress.

  > Cannot use 'import.meta' outside a module

When Cypress detects uncaught errors originating from your application it will automatically fail the current test.

This behavior is configurable, and you can choose to turn this off by listening to the `uncaught:exception` event.

https://on.cypress.io/uncaught-exception-from-application
```

### Environment

```
>  NX  Report complete - copy this into the issue template

  Node : 14.16.0
  OS   : darwin x64
  npm  : 6.14.11
  
  nx : 13.2.4
  @nrwl/angular : 13.2.4
  @nrwl/cli : 13.2.4
  @nrwl/cypress : 13.2.4
  @nrwl/devkit : 13.2.4
  @nrwl/eslint-plugin-nx : 13.2.4
  @nrwl/express : undefined
  @nrwl/jest : 13.2.4
  @nrwl/linter : 13.2.4
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : undefined
  @nrwl/nx-cloud : undefined
  @nrwl/react : undefined
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.2.4
  @nrwl/web : undefined
  @nrwl/workspace : 13.2.4
  @nrwl/storybook : 13.2.4
  @nrwl/gatsby : undefined
  typescript : 4.4.4
  rxjs : 7.4.0
  ---------------------------------------
  Community plugins:
         @angular-architects/module-federation: 12.5.3
         @angular/animations: 13.0.3
         @angular/common: 13.0.3
         @angular/compiler: 13.0.3
         @angular/core: 13.0.3
         @angular/forms: 13.0.3
         @angular/platform-browser: 13.0.3
         @angular/platform-browser-dynamic: 13.0.3
         @angular/router: 13.0.3
         @angular-devkit/build-angular: 13.0.4
         @angular/cli: 13.0.4
         @angular/compiler-cli: 13.0.3
         @angular/language-service: 13.0.3
```

**Comment by Coly010** [comment]
This is related to an issue with Webpack compiling to use module syntax with Angular 13.

A fix for Webpack and MFEs was released in Nx 13.3 yesterday.

Could you try it again using 13.3? If you have a workspace already, make sure to run the migrations as it should patch your webpack config.

**Comment by smasala** [comment]
@Coly010 no luck. no migrations were available and even a completely fresh workspace had the same outcome

```
>  NX  Report complete - copy this into the issue template

  Node : 16.10.0
  OS   : darwin x64
  yarn : 1.22.10
  
  nx : 13.3.0
  @nrwl/angular : 13.3.0
  @nrwl/cli : 13.3.0
  @nrwl/cypress : 13.3.0
  @nrwl/devkit : 13.3.0
  @nrwl/eslint-plugin-nx : 13.3.0
  @nrwl/express : undefined
  @nrwl/jest : 13.3.0
  @nrwl/linter : 13.3.0
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : undefined
  @nrwl/nx-cloud : undefined
  @nrwl/react : undefined
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.3.0
  @nrwl/web : undefined
  @nrwl/workspace : 13.3.0
  @nrwl/storybook : 13.3.0
  @nrwl/gatsby : undefined
  typescript : 4.4.4
  rxjs : 7.4.0
  ---------------------------------------
  Community plugins:
         @angular-architects/module-federation: 13.0.1
         @angular/animations: 13.1.0
         @angular/common: 13.1.0
         @angular/compiler: 13.1.0
         @angular/core: 13.1.0
         @angular/forms: 13.1.0
         @angular/platform-browser: 13.1.0
         @angular/platform-browser-dynamic: 13.1.0
         @angular/router: 13.1.0
         @angular-devkit/build-angular: 13.0.4
         @angular/cli: 13.0.4
         @angular/compiler-cli: 13.1.0
         @angular/language-service: 13.1.0
```

**Comment by Coly010** [comment]
Hey @smasala. It's odd that there were no migrations available. Can you create a minimal repo that I could use to investigate this issue more closely to your setup?

**Comment by smasala** [comment]
@Coly010  You can recreate the error with same simple steps as above. `npx create-nx-workspace` and then add an app with the `--mfe` flag.

**Comment by Coly010** [comment]
Unfortunately, this is a result of how the Angular compilation process attaches the script tag for the styles in dev mode to the `index.html` file. 
From the testing that has been done, we don't see any erroneous behaviour as a result of this error message. And it doesn't leak into production because the styles in the app get compiled to a pure CSS file, rather than a JS file.

**Comment by smasala** [comment]
@Coly010 here's the fix https://github.com/angular-architects/module-federation-plugin/blob/main/migration-guide.md#advanced-opting-out-of-using-ecmascript-modules

Add the following to the webpack config:
`scriptType: 'text/javascript'`

**Comment by Coly010** [comment]
@smasala I've discussed this already with Manfred (the author of the Angular Architects plugin)

adding `scriptType: 'text/javascript'` is a workaround, rather than a fix. 
However, if it unblocks you, that's perfect. 

It doesn't resolve the underlying issue here which is related to the compilation to ESM by Angular and the attachment of the `styles.js` script to the `index.html` by Angular. 

I'll close this as you have been unblocked.

**Comment by PowerSupply** [comment]
Is there another issue for this since this is just a workaround and does not solve the underlying problem @Coly010 ?

**Comment by Coly010** [comment]
@PowerSupply The underlying problem comes from Cypress failing when it sees an `console.error` output in the console.

There's the workaround above to change the MFE setup, or you can tell Cypress to not fail on `console.error` reports:

https://docs.cypress.io/api/events/catalog-of-events#Uncaught-Exceptions

**Comment by Coly010** [comment]
I should update this to mention that we are working on addressing this to provide an improved out-of-box experience with MFEs.
